torch/distributed/_symmetric_memory/__init__.py

```diff
@@ -24,10 +24,14 @@ def enable_symm_mem_for_group(group_name: str) -> None:
         return
 
     group = c10d._resolve_process_group(group_name)
+    global_ranks = sorted(c10d._world.pg_group_ranks[group].keys())
+    # Different subgroups with the same name should use different stores
+    global_ranks_str = "_".join(str(r) for r in global_ranks)
     store = c10d.PrefixStore(
-        "symmetric_memory",
+        f"symmetric_memory-{global_ranks_str}",
```

Comment:
curious can't we use f"symmetric_memory-{group_name}" directly?

iirc `group_name` is unique already for all subgroups

Reply:
Oh are they? I'll check it out. Your suggestion would be better if that's the case